### ISSUE TITLE
Request for support: Smart Life Pet Feeder with camera

### ISSUE THREAD

**Issue by andersovreseth** [rospogrigio/localtuya#719]
Requesting support for Smart Life Pet Feeder with camera for this integration. I am able to discover the device but not add it. Error message is: Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists.

**Comment by okidokima** [comment]
Same problem here, is there a workaround available, e.g. adding the device manually?

**Comment by Cheerpipe** [comment]
+1

**Comment by ov1d1u** [comment]
@okidokima I managed to add my pet feeder which showed the same error like OP's.

The steps I did were:

1. Add a localtuya integration in /config/integrations
2. Check Do not configure a Cloud API account (not sure if needed)
3. Configure the integration, select Add a new device
4. Select your pet feeder from the list
5. Replace device id with the one returned by `python -m tinytuya wizard` on the `id` key (see [tinytuya](https://pypi.org/project/tinytuya/) for details)
6. Fill the corresponding local key value for your device. For the protocol version I left 3.3 selected
7. For manual DPS my feeder uses 101 for the status light. There are some more, you can see [the Zigbee2MQTT documentation](https://www.zigbee2mqtt.io/advanced/support-new-devices/03_find_tuya_data_points.html) about discovering DP ids for Tuya devices. There's also 245 which is used for triggering the feeding motor, but we'll leave this for later
8. In the next step, configure 101 as a switch. So you first select _switch_ in the Entity type selection dialog, leave 101 selected, give it a name and confirm.
9. Now you should have a new device under the LocalTuya configuration with a switch entity which controls the indicator light status. To trigger the motor, you call the `localtuya.set_dp` service with the following data:

```yaml
device_id: bf9da2901bc2f8614dtnlt
dp: 245
value: 2
```
where `value` is the number of portions. You can then create a script or automation to call that whenever you'd like.

I went in a bit more details [here](https://blog.ov1d1u.net/2022/11/configure-tuya-pet-feeder-with-home.html), let me know if it worked for you or you need any help.

**Comment by pergolafabio** [comment]
Guys, also want a pet feeder, is there a compatible one with camera support? My intention was to buy one from petlibro

**Comment by ov1d1u** [comment]
@pergolafabio My Grunluft PP003, and I believe this is the same for all the other Tuyas, has a camera with RTSP streaming, so you should be able to integrate it in whathever platform that supports this protocol.

**Comment by pergolafabio** [comment]
Ah, it exposes an rtsp stream? Is it coming from the device itself? Or from the tuya integration in HA? 

I can add it then as extra Camara in my surveillance system

**Comment by ov1d1u** [comment]
@pergolafabio Yes, for my device the RTSP address it's something like `rtsp://admin:admin@192.168.0.123:554/1`. Credentials weren't set by me, are the default from the device and I don't have any idea if they can be changed, so you might want to make sure that the 554 port is not available from the internet.

**Comment by pergolafabio** [comment]
Ah cool, how did you figure out, with a Wireshark trace?

But I was googling, seems the grunlift doesn't have 2 feeders, I need one with 2 feeders, my cats always eat together :+)

**Comment by ov1d1u** [comment]
I found the information somewhere on the internet. As I said, I suspect that all Tuyas are working the same, but I can't guarantee for that.

**Comment by pergolafabio** [comment]
Yeah, don't think the rtsp has nothing todo with tuya itself

**Comment by pergolafabio** [comment]
Hey @ov1d1u ..

Is it possible to share a screenshot of your localtuya config and maybe your Lovelace cards/config ?

**Comment by ov1d1u** [comment]
@pergolafabio This is my configuration.

![Screenshot_20221203-150525](https://user-images.githubusercontent.com/1385202/205443012-776a582c-7051-493e-956a-64f8752a15b8.png)

![Screenshot_20221203-150746](https://user-images.githubusercontent.com/1385202/205443187-68f666f5-fbf3-404d-8dd4-902f696f5969.png)


`value` is the number of portions to be fed.

**Comment by pergolafabio** [comment]
Do you mind sharing those entities too? :-)

**Comment by ov1d1u** [comment]
Do you mean this?

![Screenshot_20221203-154641](https://user-images.githubusercontent.com/1385202/205444383-71846228-4c8a-4ce4-8a52-27486808a09a.png)

**Comment by pergolafabio** [comment]
No, when you add that device to tuya, afterwards you created those entities with those specific values, like 102,239... Is there no config behind them? Like min/max value and stuff like that...

**Comment by ov1d1u** [comment]
Sorry for the silence, I wasn't at home when I was replying to those comments then I forgot about it.

No, there's no config like that. Since the device is not supported, localtuya doesn't even find any DPs, so you have to add them all by yourself. It's more like a blind integration of the device.

**Comment by pergolafabio** [comment]
ok, lets try when it arrives from Ali :-)

thnx in advance

**Comment by pergolafabio** [comment]
so guys, my petfeeder arrived , is it already possible to capture the camera from it?

**Comment by ov1d1u** [comment]
@pergolafabio Try rtsp://admin:admin@<local_ip>:554/1

**Comment by pergolafabio** [comment]
Also got this error, dont think i can use localtuya for now :-(

![image](https://user-images.githubusercontent.com/43251136/206680652-1c6a277b-f019-46b5-86e4-6c70311f95d3.png)

**Comment by ov1d1u** [comment]
@pergolafabio This is normal, see https://blog.ov1d1u.net/2022/11/configure-tuya-pet-feeder-with-home.html for instructions about how to manually configure the datapoints.

**Comment by pergolafabio** [comment]
aha, seems the indicator light is already working indeed :-)
seems 245 DP for service portions, is not working yet, gonna debug later on the IOT platform to hopefull figure it out...
can you maybe show a sreenshot how it looks like on the IOT platform ?

thnx!!!!

**Comment by pergolafabio** [comment]
do you also think i can get the camera somehow?

**Comment by pergolafabio** [comment]
ok @ov1d1u , found the debug part, but where do you see the DP ??

![image](https://user-images.githubusercontent.com/43251136/206695306-9041b63f-42ca-4006-80e8-a6d109281f7f.png)

**Comment by pergolafabio** [comment]
hey @ov1d1u , can you help me obtaining the dev id? where do you see it on the cloud platform? according to your blog, you also had a chinese text for the DP IP , but how do you see the ID itself ?

![image](https://user-images.githubusercontent.com/43251136/206866230-7ec2601a-fa63-4815-9f1e-6c21f03500ad.png)

**Comment by pergolafabio** [comment]
For the ones with same problem as me to find all DP's that are not visible on SDK from tuya itself, use this:

https://github.com/rospogrigio/localtuya/issues/1188

**Comment by nyok92** [comment]
Here are my findings for [Momo's choice smart pet feeder with camera ](https://www.amazon.fr/Momos-Choice-Distributeur-Automatique-bidirectionnelle/dp/B099PRX9YM)

rtsp://ip:554
Video stream ok, but audio is very noisy (perhaps a protection or whatsoever)
Audio stream through tuya smart is clean

101-switch: Indicator light  [True/False]
104-switch: Timestamp  [True/False]
108-select: Night vision "range": ["0"auto,"1"off,"2"on] 
109-Sensor: SD card capacity (string) string "maxlen": 255

110: SD card Status (report) (integer min 1 max 5 scale 0 step 1)
111: Format SD card (report) (ON/OFF)
113: Recording mode: Record when feeding (publish)  [True/false]
117: Formatted state (report)  "min": -20000,"max": 200000,"scale": 0,"step": 1

150-switch : Record to SD card [True/False]                
151-select: Recording mode "range": ["1"Record events only,"2"continuous recording]

162-switch: Restart [True/False] 
232: weight for 1 portion ?
233: control command ??
234: peripheral status ???
235: regular feeding ??
236: historical report
237-localtuya.set_dp: Regular Feeding setting (set time and portion) (through Service localytuya.set_dp)
      cmd:7f08000310000007f12000310000007f19000310000007f1431051000000
           7f/HH:MM/portions/1000000 -->1st setting , 7f/HH:MM/portions/1000000 --> n settings

238-sensor: Battery [High, low]
239-binary sensor: Food bucket ["Full", ???]
240-binary_sensor : Jam State
241-binary_sensor : stuck state

242: feeding record ???
243: Alarm manual feeding report
244: Alar regular feeding report
245-localtuya.set_dp: Manual feed (through Service localytuya.set_dp)
246: power report ?
247: Transparent feeding transmission ????

**Comment by superpower10** [comment]
How to trigger an manual feeding from Home Assistant?

I want to install a physical button in the kitchen which feeds as soon as I press that button.

**Comment by ov1d1u** [comment]
You call `localtuya.set_dp` service like that:

```
service: localtuya.set_dp
data:
  device_id: your-device-id
  dp: 245
  value: 1
```

where `value: 1` is the number of portions.

**Comment by framerate** [comment]
Anyone found a mapping for PLAF103 Wifi Feeder yet?
![image](https://github.com/rospogrigio/localtuya/assets/74291/b75c4695-1bdb-4747-83ae-cce403f3f4eb)

**Comment by framerate** [comment]
PLAF203 `manual_feed` is 232.

Save the next person the 4 hours it took me to figure it out <3

**Comment by shedman214** [comment]
I need help configuring my pet feeder to use its entities. adding it is kinda confusing?  
![image](https://github.com/rospogrigio/localtuya/assets/32603273/a87f933a-248e-431d-a786-3799f33890dc)

**Comment by CloCkWeRX** [comment]
https://developer.tuya.com/en/docs/iot/categorycwtswsq?id=Kaiuz24lq3fq5 annoyingly doesn't marry up with what people are seeing

**Comment by 6254sagebrush** [comment]
Thanks all for this amazing tips.
One question: is there a way to add a binary_sensor for the motion detection ?
